tools/integration_tests/multiwriters_and_multimounts/multiwriters_from_two_mounts_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multiwriters_and_multimounts_test
+
+import (
+	"os"
+	"path"
+	"strings"
+	"syscall"
+	"testing"
+
+	operations "github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+	. "github.com/jacobsa/ogletest"
+)
+
+func getMountPath1() string {
+	return path.Join(setup.MntDir(), MountName1)
+}
+
+func getMountPath2() string {
+	return path.Join(setup.MntDir(), MountName2)
+}
+
+func getFilePathInMount1() string {
+	return path.Join(getMountPath1(), CommonFileName)
+}
+
+func getFilePathInMount2() string {
+	return path.Join(getMountPath2(), CommonFileName)
+}
+
+func getGCSObjPath() string {
+	return path.Join(setup.TestBucket(), CommonFileName)
+}
+
+func openFile(t *testing.T, filePath string) (file *os.File) {
+	file, err := os.OpenFile(filePath, os.O_RDWR|syscall.O_DIRECT, operations.FilePermission_0600)
+	if err != nil {
+		t.Errorf("error in opening file %s: %v", file.Name(), err)
+	}
+	return
+}
+
+func writeToFile(t *testing.T, file *os.File, fileContent string) {
+	_, err := file.WriteString(fileContent)
+	if err != nil {
+		t.Errorf("error in writing to the file %s: %v", file.Name(), err)
+	}
+}
+
+func syncFile(t *testing.T, file *os.File) {
+	err := file.Sync()
+	if err != nil {
+		t.Errorf("error in syncing file %s: %v", file.Name(), err)
+	}
+}
+
+func cleanAndCreateCommonFileInTestBucket(t *testing.T) {
+	// Cleaning one mount is sufficient as the test bucket is common.
+	err := operations.EmptyDir(getMountPath1())
+	if err != nil {
+		t.Errorf("Error while cleaning mount %v", err)
+	}
+
+	// Create file in mount 1.
+	_, err = os.Create(getFilePathInMount1())
+	if err != nil {
+		t.Errorf("Error creating common file %v", err)
+	}
+	// List files in mount 2 so that the common file starts reflecting in mount 2.
```

Comment:
I didnt get this. why do you need to do os.ReadDir() on mountPath2. Any file operation should start showing new files in mountPath2

Reply:
Reason for this:
 - We are using default stat cache capacity.
 - The StatObject call during LookUpInode that happens before opening file is [here](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/inode/dir.go#L291). As you can see we are not forcing to go to GCS for this stat call. So if stat cache has the object information then it will not go to GCS.
 - The stat cache contains generation of the backing object of file even though we haven't opened/stat the file because stat cache can also be populated at the time of ReadDir call ([here](https://github.com/jacobsa/gcloud/blob/master/gcs/gcscaching/fast_stat_bucket.go#L225))
 - The ReadDir call for writer 2 (W2)'s mount (mountPath2) was done here before W1 flushed the file and hence opening the file in mountPath2 would open file with stale generation file which will throw object not found at the time of writing the file.
 
 We can avoid this by using --stat-cache-capacity 0. Let me know if that looks better.